include/rdma/fi_log.h

```diff
@@ -35,8 +35,6 @@
 #ifndef FI_LOG_H
 #define FI_LOG_H
 
-#include "config.h"
```

Comment:
`fi_log.h` is not documented as being user-visible; that's why I concluded that it is not user-visible.  If it *is* supposed to be user-visible, then I agree: it should not include `config.h` -- there are standard workarounds for getting configury output in user-visible header files (e.g., OMPI does this kind of thing).

Is it an explicit goal to be able to build providers out-of-tree?  If so, is it known if that actually works?